### ISSUE TITLE
[Accessibility] Selectable Menu Items in Dropdown missing aria markup

### ISSUE THREAD

**Issue by donnieberg** [salesforce/design-system-react#1709]
Markup should include `aria-checked` and `role="menuitemchecked"` - essentially, they should match the SLDS markup: https://latest-218.lightningdesignsystem.com/components/menus/#With-Selectable-Items
